<?xml version="0.0" ?>
<robot name="hector_v2">
	<link name="torso">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<mass value="4.98000"/>
			<inertia ixx="0.048637" ixy="0.000154" ixz="0.000534" iyy="0.04267" iyz="0" izz="0.023157"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/torso.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/torso.stl" scale="1.00000 1.00000 1.00000" />
			</geometry>
			<material name="space_grey">
                <color rgba="0.4 0.4 0.4 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_hip1">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.05147 0.01453 -0.06160"/>
			<mass value="0.77833"/>
			<inertia ixx="0.00122" ixy="0" ixz="0" iyy="0.00111" iyz="0" izz="0.00081"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_hip1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_hip1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/>
            </material>
		</visual>	
	</link>
	<link name="l_hip2">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="-0.05690 -0.00009 0.00023"/>
			<mass value="0.67724"/>
			<inertia ixx="0.00053" ixy="0" ixz="0" iyy="0.00087" iyz="0" izz="0.00061"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_hip2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_hip2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_thigh">
		<inertial>
			<origin rpy="0.00000 0.05702 1.57080" xyz="0.00000 0.04955 -0.05388"/>
			<mass value="1.55800"/>
			<inertia ixx="0.00641282" ixy="0" ixz="0" iyy="0.00621905" iyz="0" izz="0.00163248"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_thigh.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_thigh.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_calf">
		<inertial>
			<origin rpy="0.00000 -0.00000 1.57080" xyz="0.00204 0.00000 -0.07700"/>
			<mass value="0.14600"/>
			<inertia ixx="0.00075" ixy="0" ixz="0" iyy="0.00074" iyz="0" izz="0.00002"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/calf.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/calf.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_toe">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00246 0.00003 -0.02248"/>
			<mass value="0.08440"/>
			<inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00011" iyz="0" izz="0.00010"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/foot.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/foot.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_hip1">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.05147 -0.01453 -0.06160"/>
			<mass value="0.77833"/>
			<inertia ixx="0.00122" ixy="0" ixz="0" iyy="0.00111" iyz="0" izz="0.00081"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_hip1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_hip1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_hip2">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="-0.05690 0.00009 0.00023"/>
			<mass value="0.67724"/>
			<inertia ixx="0.00053" ixy="0" ixz="0" iyy="0.00087" iyz="0" izz="0.00061"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_hip2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_hip2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_thigh">
		<inertial>
			<origin rpy="0.00000 -0.05811 1.57080" xyz="0.00000 -0.04955 -0.05343"/>
			<mass value="1.55800"/>
			<inertia ixx="0.0064221" ixy="0" ixz="0" iyy="0.00622769" iyz="0" izz="0.00163184"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 -0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_thigh.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 -0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_thigh.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_calf">
		<inertial>
			<origin rpy="0.00000 -0.00000 1.57080" xyz="0.00204 0.00000 -0.07700"/>
			<mass value="0.14600"/>
			<inertia ixx="0.00075" ixy="0" ixz="0" iyy="0.00074" iyz="0" izz="0.00002"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/calf.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/calf.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_toe">
		<inertial>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00246 -0.00003 -0.02248"/>
			<mass value="0.08440"/>
			<inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00011" iyz="0" izz="0.00010"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/foot.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/foot.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_shoudler1">
		<inertial>
			<origin rpy="0.00000 -0.00000 1.57080" xyz="0.00071 0.02838 0.04835"/>
			<mass value="0.43560"/>
			<inertia ixx="0.00043" ixy="0" ixz="0" iyy="0.00035" iyz="0" izz="0.00027"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_shoulder1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_shoulder1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_shoulder2">
		<inertial>
			<origin rpy="1.57080 -0.00000 0.13030" xyz="-0.02940 0.04541 0.00089"/>
			<mass value="0.43200"/>
			<inertia ixx="0.000422621" ixy="0" ixz="0" iyy="0.00033" iyz="0" izz="0.000267379"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_shoulder2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_shoulder2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_upper_arm">
		<inertial>
			<origin rpy="0.01136 0.13049 1.65424" xyz="0.04492 0.01006 -0.03218"/>
			<mass value="0.66484"/>
			<inertia ixx="0.00329956" ixy="0" ixz="0" iyy="0.00317918" iyz="0" izz="0.000441259"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_upper_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/l_upper_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="l_lower_arm">
		<inertial>
			<origin rpy="0.00000 -0.00000 1.57080" xyz="0.00000 0.00008 -0.11722"/>
			<mass value="0.15780"/>
			<inertia ixx="0.0013" ixy="0" ixz="0" iyy="0.00129" iyz="0" izz="0.00002"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/lower_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/lower_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>

	</link>
	<link name="r_shoulder1">
		<inertial>
			<origin rpy="0.00000 -0.00000 1.57080" xyz="0.00071 -0.02838 0.04835"/>
			<mass value="0.43560"/>
			<inertia ixx="0.00043" ixy="0" ixz="0" iyy="0.00035" iyz="0" izz="0.00027"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_shoulder1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_shoulder1.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_shoulder2">
		<inertial>
			<origin rpy="1.57080 -0.00000 -0.13030" xyz="-0.02940 -0.04541 0.00089"/>
			<mass value="0.43200"/>
			<inertia ixx="0.000422621" ixy="0" ixz="0" iyy="0.00033" iyz="0" izz="0.000267379"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_shoulder2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_shoulder2.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_upper_arm">
		<inertial>
			<origin rpy="0.01136 -0.13049 1.48735" xyz="0.04492 -0.01006 -0.03218"/>
			<mass value="0.66484"/>
			<inertia ixx="0.00329956" ixy="0" ixz="0" iyy="0.00317918" iyz="0" izz="0.000441259"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_upper_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/r_upper_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<link name="r_lower_arm">
		<inertial>
			<origin rpy="0.00000 -0.00000 1.57080" xyz="0.00000 0.00008 -0.11722"/>
			<mass value="0.15780"/>
			<inertia ixx="0.0013" ixy="0" ixz="0" iyy="0.00129" iyz="0" izz="0.00002"/>
		</inertial>
		<collision>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/lower_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
		</collision>
		<visual>
			<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 0.00000"/>
			<geometry>
				<mesh filename="package://hector_mujoco/assets/hector_v2/meshes/lower_arm.stl" scale="1.00000 1.00000 1.00000"/>
			</geometry>
			<material name="black">
                <color rgba="0.25 0.25 0.25 1"/> <!-- Red with 50% transparency -->
            </material>
		</visual>
	</link>
	<joint name="l_hip_yaw" type="revolute">
		<parent link="torso"/>
		<child link="l_hip1"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.04700 -0.12650"/>
		<limit effort="33.5" lower="-0.78540" upper="0.78540" velocity="21.0"/>
		<axis xyz="0.00000 0.00000 1.00000"/>
	</joint>
	<joint name="l_hip_roll" type="revolute">
		<parent link="l_hip1"/>
		<child link="l_hip2"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.04650 0.01500 -0.07050"/>
		<limit effort="33.5" lower="-0.78540" upper="0.78540" velocity="21.0"/>
		<axis xyz="1.00000 0.00000 0.00000"/>
	</joint>
	<joint name="l_hip_pitch" type="revolute">
		<parent link="l_hip2"/>
		<child link="l_thigh"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="-0.06000 0.01800 0.00000"/>
		<limit effort="33.5" lower="-0.2984513" upper="2.099997609" velocity="21.0"/>
		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>
	<joint name="l_knee" type="revolute">
		<parent link="l_thigh"/>
		<child link="l_calf"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="-0.00000 0.01420 -0.22000"/>
		<limit effort="67.0" lower="-2.000007696" upper="0.0" velocity="10.5"/>
		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>
	<joint name="l_ankle" type="revolute">
		<parent link="l_calf"/>
		<child link="l_toe"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 -0.22000"/>
		<limit effort="33.5" lower="-1.5708" upper="1.5708" velocity="21.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>
	<joint name="r_hip_yaw" type="revolute">
		<parent link="torso"/>
		<child link="r_hip1"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.04700 -0.12650"/>
		<limit effort="33.5" lower="-0.78540" upper="0.78540" velocity="21.0"/>
		<axis xyz="0.00000 0.00000 1.00000"/>
	</joint>
	<joint name="r_hip_roll" type="revolute">
		<parent link="r_hip1"/>
		<child link="r_hip2"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.04650 -0.01500 -0.07050"/>
		<limit effort="33.5" lower="-0.78540" upper="0.78540" velocity="21.0"/>
		<axis xyz="1.00000 0.00000 0.00000"/>
	</joint>
	<joint name="r_hip_pitch" type="revolute">
		<parent link="r_hip2"/>
		<child link="r_thigh"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="-0.06000 -0.01800 0.00000"/>
		<limit effort="33.5" lower="-0.2984513" upper="2.099997609" velocity="21.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>
	<joint name="r_knee" type="revolute">
		<parent link="r_thigh"/>
		<child link="r_calf"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="-0.00000 0.00000 0.00000" xyz="-0.00000 -0.01420 -0.22000"/>
		<limit effort="67.0" lower="-2.000007696" upper="0.0" velocity="10.5"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>
	<joint name="r_ankle" type="revolute">
		<parent link="r_calf"/>
		<child link="r_toe"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.00000 -0.22000"/>
		<limit effort="33.5" lower="-1.5708" upper="1.5708" velocity="21.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>

	<joint name="l_shoulder_yaw" type="revolute">
		<parent link="torso"/>
		<child link="l_shoudler1"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.05300 0.12400"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="0.00000 0.00000 1.00000"/>
	</joint>

	<joint name="l_shoulder_pitch" type="revolute">
		<parent link="l_shoudler1"/>
		<child link="l_shoulder2"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 0.04900 0.05300"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>

	<joint name="l_shoulder_roll" type="revolute">
		<parent link="l_shoulder2"/>
		<child link="l_upper_arm"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 0.00000 0.00000" xyz="-0.01100 0.04570 0.00000"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="-1.00000 0.00000 0.00000"/>
	</joint>

	<joint name="l_elbow" type="revolute">
		<parent link="l_upper_arm"/>
		<child link="l_lower_arm"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.04100 -0.00700 -0.20000"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>

	<joint name="r_shoulder_yaw" type="revolute">
		<parent link="torso"/>
		<child link="r_shoulder1"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.05300 0.12400"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="0.00000 0.00000 1.00000"/>
	</joint>

	<joint name="r_shoulder_pitch" type="revolute">
		<parent link="r_shoulder1"/>
		<child link="r_shoulder2"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 0.00000" xyz="0.00000 -0.04900 0.05300"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>

	<joint name="r_shoulder_roll" type="revolute">
		<parent link="r_shoulder2"/>
		<child link="r_upper_arm"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="0.00000 -0.00000 -0.00000" xyz="-0.01100 -0.04570 0.00000"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="-1.00000 0.00000 0.00000"/>
	</joint>

	<joint name="r_elbow" type="revolute">
		<parent link="r_upper_arm"/>
		<child link="r_lower_arm"/>
		<dynamics damping="1.0" friction="0.0001"/>
		<origin rpy="-0.00000 -0.00000 0.00000" xyz="0.04100 0.00700 -0.20000"/>
		<limit effort="17.0" lower="-1.5708" upper="1.5708" velocity="32.0"/>

		<axis xyz="0.00000 1.00000 0.00000"/>
	</joint>

</robot>
